MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -319,14 +322,16 @@ - (void)onProcessingLog:(id<MSLog>)log withPriority:(MSPriority)priority {
   @synchronized(self) {
     NSData *serializedLog = [NSKeyedArchiver archivedDataWithRootObject:log];
     if (serializedLog && (serializedLog.length > 0)) {
-      if (self.bufferIndex > (ms_crashes_log_buffer_size - 1)) {
-        self.bufferIndex = 0;
+      if ([[self.bufferIndex objectForKey:@(priority)] integerValue] > (ms_crashes_log_buffer_size - 1)) {
+        // Reset the counter to 0 for a priority.
+        [self.bufferIndex setObject:@0 forKey:@(priority)];
       }
-      msCrashesLogBuffer[self.bufferIndex].buffer =
+      NSInteger index = [[self.bufferIndex objectForKey:@(priority)] integerValue];
+      msCrashesLogBuffer[priority][index].buffer =
           std::string(&reinterpret_cast<const char *>(serializedLog.bytes)[0],
                       &reinterpret_cast<const char *>(serializedLog.bytes)[serializedLog.length]);
-
-      self.bufferIndex += 1;
+      index += 1;
+      [self.bufferIndex setObject:@(index) forKey:@(priority)];
```

Comment:
I guess YMMV applies here. 
I think with `self.bufferIndex[@(priority)] = @(++index);`, I'd add a comment to point out the increment to make it less likely that someone doesn't see it. In this case, I can also keep `index += 1`. ;)